### ISSUE TITLE
Angular 9 support fails due to static access to ViewChild

### ISSUE THREAD

**Issue by oliverguenther** [ui-router/angular#728]
In trying to update to Angular 9, I was delighted to see ui-router already merged some Ivy support a while back.

However trying to upgrade myself, routing fails due to the UiView directive accessing a `@ViewChild` in its `ngOnInit` flow, exactly in this line:

https://github.com/ui-router/angular/blob/master/src/directives/uiView.ts#L305

Here, `_componentTarget` is still undefined and cannot be accessed. A solution would be to move this block into an `ngAfterViewInit` flow instead.

Do you know about this issue and would welcome PRs towards it, or do you have a better suggestion?

**Comment by christopherthielen** [comment]
I'm not aware of this issue.  Can you point to the v9 docs that explain the differences?  

For what it's worth, the current build passes CI with Angular v9: 

- https://github.com/ui-router/angular/runs/485609823?check_suite_focus=true
- [Raw logs](https://pipelines.actions.githubusercontent.com/BsjhEYVNUhWrnWRy5FKpQ09QPtUhyJ7ulCmhtEbKNNqwtkig3w/_apis/pipelines/1/runs/69/signedlogcontent/7?urlExpires=2020-03-04T18%3A03%3A05.3127169Z&urlSigningMethod=HMACV1&urlSignature=AqMrmFOeaWKuQZvmDETyADJ61A8VOgF0KUib4Jyd9GE%3D)

and the sample app has been updated to Angular v9:

- https://github.com/ui-router/sample-app-angular/blob/master/package.json#L17-L22

**Comment by oliverguenther** [comment]
While creating the PR, I too saw that it appears to work fine with the test. I stumbled upon this error while debugging errors in our own application upgrade to Angular 9 that uses ui-router.

I was remembering to read about the ng migration script for version 9 removing all the ViewChild flags and thinking that it was deprecated (The default [was changed in v8](https://angular.io/guide/static-query-migration#what-does-this-flag-mean) and I remembered wrongly).

Turns out we were not running the latest version of ui-router/core and /angular. Updating to the latest version works fine. Sorry for the confusion and thanks for your time and effort!
